### DISCUSSION TITLE
Brainstorm - How could Plausible track retention / loyalty

### DISCUSSION THREAD

**Discussion by sptv2reg** [plausible/analytics#2594] [Anything Else]
Hi

We're considering using Plausible for our media sites. It really is wonderfully simple compared to GA4, but there are obviously also some limitations. One of them is adding metadata to pageviews which I understand is in the pipeline (great).

But the big one is tracking users beyond 24 hours. Loyalty is such an important metric for news websites and it doesn't seem possible to track this with Plausible out-of-the-box.

I think it's important to keep the no-cookie approach, but I was thinking if there was a way to track things like:

Sessions pr. User (daily)
Sessions pr. User (weekly)
Sessions pr. User (monthly)

without sending any PII whatsoever to the server. One approach could be to gather these statistics clientside (in a cookie, but without any PII whatsoever) and then dispatch as an event to plausible using custom events.

I would simply increment a counter and store the value in a cookie (also with some timestamp), and then whenever a given interval was exceeded (and the visitor came around the website again), I would send the count to Plausible. So I would end up with data points indicating the numbers for completely anonymous visitors. I could then calculate an average from them.

Obviously this wouldn't be completely precise, but I think it would be enough (for high traffic websites) to get some ideas whether for example we managed to increase the rate of visitors coming back to our site.

What do you think - would something like this work? Or are there any other approaches I haven't thought about?

**Comment by metmarkosaric** [comment]
hi @sptv2reg! 

any long term retention is difficult due to us being privacy-first, not having any long-term identifiers and trying to be respectful of the privacy laws without requiring you to put our script behind a consent form. 

"Sessions pr. User (daily)" already works with the way Plausible works now by the way as pretty much everything we do is session based. you can get the number of unique visitors for any given day directly in our dashboard and we currently also provide the number of sessions on any given day in [our stats API](https://plausible.io/docs/stats-api#metrics). so taking the number of sessions and dividing it by the number of unique visitors will give you the daily number of sessions per user. we plan on introducing the number of sessions metric directly in the dashboard too at some point soon. stay tuned!
